adapter.go

```diff
@@ -264,6 +267,18 @@ func TurnOffAutoMigrate(db *gorm.DB) *gorm.DB {
 }
 
 func NewAdapterByDBWithCustomTable(db *gorm.DB, t interface{}, tableName ...string) (*Adapter, error) {
+
+	r := reflect.ValueOf(t).Type()
+	if r.Kind() == reflect.Ptr {
+		r = r.Elem()
+	}
+	for _, field := range []string{"ID", "Ptype", "V0", "V1", "V2", "V3", "V4", "V5", "V6", "V7"} {
```

Comment:
why are we forcing the column names to be one from this list? If it's a custom table, can we not let users define their own column names?

Reply:
For GORM autodelete to work, the name of the column has to be `deletedat`. Which means the custom table will not support GORM autodelete